### ISSUE TITLE
Simplify the plugin development documentation

### ISSUE THREAD

**Issue by davidhicks980** [rollup/rollup#4129]
Documentation Is:

- [ ] Missing
- [ ] Needed
- [x] Confusing
- [ ] Not Sure?

### Please Explain in Detail...
Hey Rollup team! First off, I love Rollup so thanks for what you all do. Second, I'm just now writing a rollup plugin and I find the simple example to be a little abstract. Let me show you what I mean: 

**A Simple Example**
The following plugin will intercept any imports of virtual-module *for newcomers who are unaware of what virtual-module is, what does it mean?* without accessing the file system. This is for instance necessary if you want to use Rollup in a browser *how so?*.
```
// rollup-plugin-my-example.js
export default function myExample () {
  return {
    name: 'my-example', // this name will show up in warnings and errors
    resolveId ( source ) {                 /*what is an ID? What is the source?*/
      if (source === 'virtual-module') {
        return source; // this signals that rollup should not ask other plugins /*what do you mean by asking other plugins?*/ or check the file system to find this id
      }
      return null; // other ids should be handled as usually /*what is the usual case?*/
    },
    load ( id ) {
      if (id === 'virtual-module') {
        return 'export default "This is virtual!"'; // the source code for "virtual-module"
      }
      return null; // other ids should be handled as usually
    }
  };
}

// rollup.config.js
import myExample from './rollup-plugin-my-example.js';
export default ({
  input: 'virtual-module', // resolved by our plugin
  plugins: [myExample()],
  output: [{
    file: 'bundle.js',
    format: 'es'
  }]
});
```
### Your Proposal for Changes
It may be helpful to have the simple example be more concrete. For example, showing how to create a plugin that adds a comment to a file and prefixes the file after it has been transpiled with SASS. This way, readers would get an understanding of how to correctly fit their plugin into the bundling process. As soon as I grasp the transpilation process a little better, I'll try to create an example that demonstrates what I mean.

**Comment by lukastaegert** [comment]
Maybe we should have more than one "simple example". I would still like to include the current introduction example as it is a very important use case that is vital to understand. But then again, maybe a few more explanatory words would help as well. In any case, docs PRs are always welcome!

**Comment by davidhicks980** [comment]
Yeah I think that sounds great. I figured there was a good reason to
include the example given. I'll read through the docs and try to write
another example to add on.
